### PR TITLE
Deprecate async SFTP sensor

### PR DESCRIPTION
**Pull request by pankajkoti**
Deprecate SFTPSensorAsync and proxy it to its Airflow
OSS provider's counterpart

closes: #1419

### CODE REVIEW

**Comment by pankajkoti** [comment]
This needs to wait for the upstream provider PR to be released https://github.com/apache/airflow/pull/37117

**Review by phanikumv — APPROVED** [review]

**Review by pankajastro — APPROVED** [review]
LGTM, once upstream provider version updated here

File: setup.cfg [hunk 1]
```diff
@@ -69,7 +69,8 @@ http =
 microsoft.azure =
     apache-airflow-providers-microsoft-azure>=9.0.0
 sftp =
-    apache-airflow-providers-sftp
+    # Increment to the upcoming release version containing the deferrable SFTP support PR https://github.com/apache/airflow/pull/37117
+    apache-airflow-providers-sftp>=4.9.0rc1
```
Comment (pankajkoti) [inline setup.cfg:73]:
```suggestion
    apache-airflow-providers-sftp>=4.9.0
```

File: setup.cfg [hunk 2]
```diff
@@ -126,7 +127,8 @@ all =
     apache-airflow-providers-google>=10.14.0
     apache-airflow-providers-http>=4.9.0
     apache-airflow-providers-snowflake>=5.3.0
-    apache-airflow-providers-sftp
+    # Increment to the upcoming release version containing the deferrable SFTP support PR https://github.com/apache/airflow/pull/37117
+    apache-airflow-providers-sftp>=4.9.0rc1
```
Comment (pankajkoti) [inline setup.cfg:131]:
```suggestion
    apache-airflow-providers-sftp>=4.9.0
```

**Review by Lee-W — APPROVED** [review]

**Pull request merged**